frontend/src/components/CompanyInput.tsx

```diff
@@ -22,10 +26,26 @@ const CompanyInput: React.FC<CompanyInputProps> = ({ onCompanyNameSubmit, onComp
     debouncedOnChange(name);
   };
 
-  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
     if (companyName.trim() !== '') {
-      onCompanyNameSubmit(companyName.trim()).then(() => setCompanyName(''));
+      try {
```

Comment:
As discussed in the chat, user must not be able to leave a review for more than one company, regardless of equality checks, otherwise the logic to compute the rating becomes too hard to be adapted.